internal/core/rule/actions/api.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package actions
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/url"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/erda-project/erda/internal/core/rule/jsonnet"
+	"github.com/erda-project/erda/pkg/http/httpclient"
+)
+
+type APIConfig struct {
+	TemplateParser jsonnet.Engine
+	URL            string
+	Header         Header
+	Body           string
+	Snippet        string
+}
+
+type Header struct {
+	Key   string
+	Value string
+}
+
+// outgoing API
+func (a *APIConfig) Send(content map[string]interface{}) (string, error) {
```

Comment:
add comment for content: what's k,v represented?

Reply:
add add comments for a block logic(step)

Reply:
done